Dockerfile

```diff
@@ -2,20 +2,16 @@ FROM nginx:1.23.3-alpine
 
 LABEL maintainer="fehguy"
 
-ENV BASE_URL ""
+ENV BASE_URL "/" \
+    NGINX_PORT "8080"
 
-COPY nginx.conf /etc/nginx/
+COPY nginx.conf /etc/nginx/templates/default.conf.template
 
 COPY ./index.html /usr/share/nginx/html/
 COPY ./dist/oauth2-redirect.html /usr/share/nginx/html/
 COPY ./dist/* /usr/share/nginx/html/dist/
-COPY ./docker-run.sh /usr/share/nginx/
+COPY ./docker-run.sh /docker-entrypoint.d/91-docker-run.sh
 
-
-RUN chmod +x /usr/share/nginx/docker-run.sh && \
-    chmod -R a+rw /usr/share/nginx && \
-    chmod -R a+rw /etc/nginx
+RUN chmod +x /docker-entrypoint.d/91-docker-run.sh
 
 EXPOSE 8080
-
-CMD ["sh", "/usr/share/nginx/docker-run.sh"]
```

Comment:
TECHNICAL NOT EXPLAINING THE CHANGE

> No need for `CMD` as `/docker-entrypoint.d/91-docker-run.sh` is used now

Reply:
This is the trick. Nginx image will [execute all scripts](https://github.com/nginxinc/docker-nginx/blob/master/mainline/alpine-slim/docker-entrypoint.sh#L16) from the `/docker-entrypoint.d` folder and [`/docker-entrypoint.d/91-docker-run.sh`](https://github.com/swagger-api/swagger-editor/pull/3958/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R13) too (instead of the old location `/usr/share/nginx/docker-run.sh`) before to run whatever provided in `CMD`.
After that default [CMD of nginx container](https://github.com/nginxinc/docker-nginx/blob/master/mainline/alpine-slim/Dockerfile#L121) will be applied that basically runs the nginx in foreground.
So we do not need to rewrite this part, but use default CMD instead, because our script was executed in line 9. And nginx afterwads up line 10:

```shell
# docker logs swagger-editor 2>&1 | nl -w 1
1       /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
2       /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
3       /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
4       10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
5       10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
6       /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
7       20-envsubst-on-templates.sh: Running envsubst on /etc/nginx/templates/default.conf.template to /etc/nginx/conf.d/default.conf
8       /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
9       /docker-entrypoint.sh: Launching /docker-entrypoint.d/91-docker-run.sh
10      2023/03/22 10:18:20 [notice] 35#35: using the "epoll" event method
11      2023/03/22 10:18:20 [notice] 35#35: nginx/1.23.3
12      2023/03/22 10:18:20 [notice] 35#35: built by gcc 12.2.1 20220924 (Alpine 12.2.1_git20220924-r4) 
13      2023/03/22 10:18:20 [notice] 35#35: OS: Linux 5.15.0-67-generic
14      2023/03/22 10:18:20 [notice] 35#35: getrlimit(RLIMIT_NOFILE): 1048576:1048576
15      2023/03/22 10:18:20 [notice] 35#35: start worker processes
16      2023/03/22 10:18:20 [notice] 35#35: start worker process 36
17      2023/03/22 10:18:20 [notice] 35#35: start worker process 37
18      2023/03/22 10:18:20 [notice] 35#35: start worker process 38
19      2023/03/22 10:18:20 [notice] 35#35: start worker process 39
```

This is also the reason, why we do not need to start nginx by our own and I [removed this line](https://github.com/swagger-api/swagger-editor/pull/3958/files#diff-360c358e5fc4de52699ff11376043b7c67ebccc164c8dd9bc2b025782d681279L67)

Reply:
Yep, this comment was for reference purposes only. No action on your side was needed.